content/posts/introducing-slight.md

```diff
@@ -0,0 +1,175 @@
+---
+title: "Introducing SpiderLightning - A Cloud System Interface with WebAssembly"
+description: "Announcing SpiderLightning and slight, a distributed application runtime for Wasm featuring portable application building blocks."
+date: 2022-10-31 00:00:00 +0000 UTC
+authorname: "Jiaxiao (Joe) Zhou"
+author: "@jiaxiao_zhou"
+authorlink: "https://twitter.com/jiaxiao_zhou"
+image: "images/logos/twitter-card.png"
+tags: ["slight", "spiderlightning", "Wasm", "WASI"]
+---
+
+At DeisLabs, we are researching and developing various WebAssembly tools and systems overtime. We have been working on a new project called SpiderLightning, which is a distributed application interface written in an Interface Definition Language (IDL) called [WIT](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md). SpiderLightning is paired with a CLI called slight, which we recently released "0.2.0" version. In this blog post, we will introduce SpiderLightning and slight.
+
+## Authoring portable distributed applications is hard
+
+Imagine you're a developer working at a consumer-facing company. You built a web app that runs on on-premises data centers and used open-source projects to build data stores and pipelines. At the beginning, there are only a few hundreds of customers, so the web app worked well. However, as the company grows, the number of customers increases rapidly. The web app becomes slow and unstable. This is a classic scaling issue. Now you're tasked to migrate from on-premises data centers to public cloud because they will horizontal scale the application for you.
```

Comment:
the first thing that will happen with this framing is that the reader will say.... uh, scaling has nothing to do with this. you just buy more computers. Why, they will ask, does the web app become slow and unstable? 

So there's a reader disconnect right out of the gate. There are MILLIONS of projects that make "web apps run faster" but you don't even mention anything ABOUT spiderlight that means "the problem of scaling is generally solved", right? 

So FRAME it a bit better in the introduction: what does WEBASSEMBLY do that containers can't do. THEN you can say hey, if you were building in containers, you would not be able to take advantage of wasm features. So the following conversation is about how if you want to take advantage of the WebAssembly component model to ease, and therefore scale out your ability to make wasm apps fast and well for the edge AND hyperscale..... something like this is required up front.

keep all this you have, of course. you'll get to it! But without the introduction that's better, stronger, and has more positioning for the reader much of the reaction will be disbelief that you couldn't have just bought more machines -- problem solved!!!

Reply:
resolved offline